[FEAT] Add integration with `huggingface_hub.utils.telemetry`

davidberenstein1957 opened the pull request:
# Description
<!-- Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

This PR adds changes to the server telemetry to gather metrics for API endpoint calls. This is the first iteration. Some new usage metrics can be included. 

The metrics gathered include the user ID and some system info as the server ID (UUID generated once when starting the Argilla server)

Also, it deprecates the old telemetry KEY ("huggingface_hub includes an helper to send telemetry data. This information helps us debug issues and prioritize new features. Users can disable telemetry collection at any time by setting the HF_HUB_DISABLE_TELEMETRY=1 environment variable. Telemetry is also disabled in offline mode (i.e. when setting HF_HUB_OFFLINE=1)."

### OUTDATED

Adds telemetry for:

- [x] users
- [x] workspaces
- [x] datasets
- [x] login users
- [x] server errors
- [x] records
- [x] responses
- [x] suggestions
- [x] metadata
- [x] vectors
- [x] deprecate old telemetry KEY ("huggingface_hub includes an helper to send telemetry data. This information helps us debug issues and prioritize new features. Users can disable telemetry collection at any time by setting the HF_HUB_DISABLE_TELEMETRY=1 environment variable. Telemetry is also disabled in offline mode (i.e. when setting HF_HUB_OFFLINE=1)."
- [x] write documentation done int #5253 
- [x] add automatic task distribution should be done after #5136
- [x] include https://github.com/gradio-app/gradio/pull/8884

General Idea:
I’ve structured data to come in through URLs/topics like `dataset/settings/vectorsettings/create` or `dateset/records/suggestions/read` along with some generalized metadata per URL/topics, like `count` or `type` of suggestion or setting.

To discuss:
- What to do with `list` methods. I currently track `list-like` and send each individual with `read`, along with a `read` with a count. I did this because it might be interesting to get the total number of users, workspaces etc. Should we move this over to `list` as a separate CRUD action? Do we also want to capture each individual update
- A similar logic applies to bulk operations. `bulk_crud` as separate CRUD actions?  
- I don't track user/dataset/workspace-specific list operations, like list_users_workspace or list_datasets_user.
- I don't track metadata and vector updates on a record level, however, we DO keep track of operations on suggestions and responses. 
- @frascuchon was there a reason to include the `header` along with `user/login` operations? otherwise I will rewrite this a bit and include the `user/login` as `user/read`.

Follow up
- https://github.com/argilla-io/argilla/issues/5224 @frascuchon_ 

Closes #5204

**Type of change**
<!--  Please delete options that are not relevant. Remember to title the PR according to the type of change  -->

- Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**
<!--  Please add some reference about how your feature has been tested.  -->
NA

**Checklist**
<!--  Please go over the list and make sure you've taken everything into account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

davidberenstein1957 committed:
Update `TelemetryClient` to use `huggingface_hub.utils`

davidberenstein1957 committed:
Update `user``CRUD` telemetry tracking

davidberenstein1957 committed:
Update `workspace` CRUD telemetry tracking

davidberenstein1957 committed:
Update `workspace` telemetry from `list_user_workspaces` method

davidberenstein1957 committed:
Fix arguments passed to `track_crud_workspace` in `list_user_workspaces`

davidberenstein1957 committed:
Fix `await` to `telemetry` call

davidberenstein1957 commented on argilla-server/src/argilla_server/telemetry.py:
```diff
@@ -17,114 +17,102 @@
             "deployment": server_deployment_type(),
             "docker": is_running_on_docker_container(),
-            "version": __version__,
         }
```
need to check if system info is needed because `send_telemetry` methods does not seem to pass it.

dvsrepo replied:
I wonder if we still need the server_id if some identification info is provided by hub_telemetry, but we can see some generated metrics with this implementation

davidberenstein1957 replied:
I agree we might limit this but the information is not passed along with the SDK request, but seems to be inferred on the server side so I was not sure about it. I do need to do some additional tests.

davidberenstein1957 commented on argilla-server/src/argilla_server/telemetry.py:
```diff
@@ -17,114 +17,102 @@
             "sys_version": platform.version(),
             "deployment": server_deployment_type(),
             "docker": is_running_on_docker_container(),
-            "version": __version__,
```
version is added within `send_telemetry`method

davidberenstein1957 replied:
@dvsrepo @frascuchon we automatically get a geo IP, but this is o a location level an not a hardware level. I did remove "python_version" since it was already being captured.

davidberenstein1957 commented on argilla-server/src/argilla_server/telemetry.py:
```diff
@@ -17,114 +17,102 @@
+    def _process_workspace_model(workspace: Workspace):
+        return {
+            "workspace_id": str(workspace.id),
+            "workspace": str(uuid.uuid5(namespace=_TELEMETRY_CLIENT.server_id, name=workspace.name)),
```
```suggestion
            "workspace_hash": str(uuid.uuid5(namespace=_TELEMETRY_CLIENT.server_id, name=workspace.name)),
```

davidberenstein1957 committed:
Update `telemetry_client: TelemetryClient = Depends(get_telemetry_client),` instead of direct `_TELEMETRY_CLIENT` import

davidberenstein1957 committed:
Add telemetry methods, dataset, workspace, user, settings, records, record_subtopic

davidberenstein1957 committed:
Add telemetry methods `fields`

davidberenstein1957 committed:
Add telemetry methods `metadata_properties`

davidberenstein1957 committed:
Add telemetry methods `questions`

davidberenstein1957 committed:
Add telemetry methods `records`

davidberenstein1957 committed:
Add telemetry methods to `responses`

davidberenstein1957 committed:
Add telemetry methods to ùsers`

davidberenstein1957 committed:
Add telemetry `suggestions`

dvsrepo commented on argilla-server/src/argilla_server/telemetry.py:
```diff
@@ -17,114 +17,195 @@
+            user_agent.update(self._process_dataset_settings(dataset=dataset))
+        self.track_data(topic=topic, user_agent=user_agent, count=count)
+
+        return None
```
Is this intended?

davidberenstein1957 replied:
Yes, I need to review and update the processing for these settings methods.

davidberenstein1957 committed:
Update `track_crud_dataset_setting` processing

davidberenstein1957 committed:
Merge branch 'develop' into feat/5204-feature-add-huggingface_hubutilssend_telemetry-to-the-argilla-server

davidberenstein1957 committed:
Add `enable_telemetry` check

davidberenstein1957 committed:
Remove `disable_send`

davidberenstein1957 committed:
Deprecate `ARGILLA_ENABLE_TELEMETRY` env var

davidberenstein1957 committed:
Update `test_telemetry`

davidberenstein1957 committed:
Add enable telemetry to post_init

davidberenstein1957 committed:
Add `UUID` to `str` covnersion

davidberenstein1957 committed:
Run tests with enabled telemetry

davidberenstein1957 committed:
Remove `telemetry` client

davidberenstein1957 committed:
Fix tests errors

davidberenstein1957 committed:
Update `test_telemetry` fixture

davidberenstein1957 committed:
Update disable telemetry env var

davidberenstein1957 committed:
Fix tests dataset creation

davidberenstein1957 committed:
Fix failing tests due to unloaded DatabaseModels

davidberenstein1957 committed:
Add tests telemetry crud datasets

davidberenstein1957 committed:
Update tests coverage telemetry tracking

davidberenstein1957 committed:
Merge branch 'develop' into feat/5204-feature-add-huggingface_hubutilssend_telemetry-to-the-argilla-server

davidberenstein1957 commented on argilla-server/src/argilla_server/settings.py:
```diff
@@ -157,9 +156,23 @@ class Settings(BaseSettings):
     # See also the telemetry.py module
-    enable_telemetry: bool = True
-    telemetry_key: str = DEFAULT_TELEMETRY_KEY
+    @validator("database_url", pre=True, always=True)
```
```suggestion
    @validator("enable_telemetry", pre=True, always=True)
```

davidberenstein1957 replied:
```suggestion
    @validator("enable_telemetry", pre=True, always=True)
```

davidberenstein1957 committed:
Update argilla-server/src/argilla_server/settings.py

davidberenstein1957 committed:
Remove Python version from sytem info

frascuchon commented on .github/workflows/argilla-frontend.deploy-environment.yml:
```diff
@@ -63,7 +63,7 @@ jobs:
             ANNOTATOR_PASSWORD=${{ steps.credentials.outputs.annotator }}
             ANNOTATOR_API_KEY=${{ steps.credentials.outputs.annotator }}
-            ARGILLA_ENABLE_TELEMETRY=0
+            HF_HUB_DISABLE_TELEMETRY=1
```
We should update docs about this var somewhere

davidberenstein1957 replied:
I did in the #5253

frascuchon reviewed (commented):
This PR has a lot of changes and is hard to track. It would be nice to deploy this on a space and see the gathered metrics.

davidberenstein1957 committed:
Update tests to also check assert call track_data

davidberenstein1957 committed:
Add documentation for telemetry information (#5253)

# Description
<!-- Please include a summary of the changes and the related issue.
Please also include relevant motivation and context. List any
dependencies that are required for this change. -->

NA

**Type of change**
<!-- Please delete options that are not relevant. Remember to title the
PR according to the type of change -->

- Documentation update

**How Has This Been Tested**
<!-- Please add some reference about how your feature has been tested.
-->

**Checklist**
<!-- Please go over the list and make sure you've taken everything into
account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

davidberenstein1957 committed:
Add async telemetry client

davidberenstein1957 committed:
Update `test_telemetry`

davidberenstein1957 commented on argilla-server/src/argilla_server/api/handlers/v1/users.py:
```diff
@@ -26,13 +25,18 @@
+    current_user: User = Security(auth.get_current_user),
+    telemetry_client: TelemetryClient = Depends(get_telemetry_client),
+):
+    await telemetry_client.track_user_login(request, current_user)
```
@frascuchon was there a reason to include the header along with user/login operations? otherwise I will simplify this a bit and include the user/login as user/read and include `is_login` as metadata.

davidberenstein1957 commented on argilla-server/src/argilla_server/api/handlers/v1/workspaces.py:
```diff
@@ -67,19 +78,25 @@ async def delete_workspace(
 
 
 @router.get("/me/workspaces", response_model=Workspaces)
 async def list_workspaces_me(
```
I don't track user/dataset/workspace-specific list operations, like list_workspaces_me.

davidberenstein1957 commented on argilla-server/src/argilla_server/telemetry.py:
```diff
@@ -17,114 +17,248 @@
+    def _process_dataset_model(dataset: Dataset):
+        return {
+            "dataset_id": str(dataset.id),
+            "dataset_hash": str(uuid.uuid5(namespace=_TELEMETRY_CLIENT.server_id, name=dataset.name)),
```
@frascuchon was there a reason to include these hashed? otherwise I would opt for just storing the ID

frascuchon replied:
The reason was to identify the same dataset names. The combination of namespace UUID + str will provide the same UUID. So, operations with dataset `my-dataset` under the same server will be identified by this hash. But this is something that we're not using. It makes sense to remove it.

davidberenstein1957 replied:
I agree, we can simplify and just reason based on IDs.

davidberenstein1957 committed:
Update list-like to basic CRUD
Fix upsert behavior suggestions

davidberenstein1957 commented on argilla-server/src/argilla_server/errors/error_handler.py:
```diff
@@ -53,15 +53,15 @@ def __init__(self, error: ServerError):
+        user_agent = {
             "code": error.code,
             "user-agent": request.headers.get("user-agent"),
             "accept-language": request.headers.get("accept-language"),
```
@frascuchon was there a reason we were including these?

frascuchon replied:
To identify API clients. Browsers send these headers which can be useful for identifying used browsers and languages

davidberenstein1957 committed:
Merge branch 'develop' into feat/5204-feature-add-huggingface_hubutilssend_telemetry-to-the-argilla-server

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

davidberenstein1957 committed:
Resolved utcnow() deprecation

davidberenstein1957 committed:
Merge branch 'feat/5204-feature-add-huggingface_hubutilssend_telemetry-to-the-argilla-server' of https://github.com/argilla-io/argilla into feat/5204-feature-add-huggingface_hubutilssend_telemetry-to-the-argilla-server

davidberenstein1957 committed:
Update to always add error type to overview

davidberenstein1957 committed:
Add dataset distribution tracking to telemetry

davidberenstein1957 committed:
Revert UTC change

davidberenstein1957 committed:
fix failing tests

davidberenstein1957 committed:
fix failing tests

davidberenstein1957 committed:
Remove server id from telemetry to be more GDPR compliant

davidberenstein1957 committed:
Update tlemetry workflow

frascuchon committed:
Merge branch 'develop' into feat/5204-feature-add-huggingface_hubutilssend_telemetry-to-the-argilla-server

davidberenstein1957 committed:
chore: add huggingface_hub to dependencies

davidberenstein1957 committed:
Merge branch 'feat/5204-feature-add-huggingface_hubutilssend_telemetry-to-the-argilla-server' of https://github.com/argilla-io/argilla into feat/5204-feature-add-huggingface_hubutilssend_telemetry-to-the-argilla-server

frascuchon commented on argilla-server/src/argilla_server/api/handlers/v1/datasets/datasets.py:
```diff
@@ -267,10 +335,7 @@ async def publish_dataset(
-        action="PublishedDataset",
-        data={"questions": list(set([question.settings["type"] for question in dataset.questions]))},
-    )
+    await telemetry_client.track_crud_dataset(action="create", dataset=dataset)
```
why not "publish" instead of "create" ?

davidberenstein1957 replied:
fine by me, I wanted to align things as much as possible within the same topic URLs, but I will differentiate a bit more and potentially also create a separate one for "list".

frascuchon commented on argilla-server/src/argilla_server/api/handlers/v1/datasets/records.py:
```diff
@@ -358,6 +364,8 @@ async def search_current_user_dataset_records(
             query_score=record_id_score_map[record.id]["query_score"],
         )
 
+    await telemetry_client.track_crud_records(action="read", record_or_dataset=dataset, count=search_responses.total)
```
Again, I would prefer to rename the action to "search" instead of "read", since we already have a "read" records endpoint. Also, adding the "me" context would be nice to separate contextualized searches (normally performed from the UI) from common searches ( normally performed from the SDK). 
```suggestion
    await telemetry_client.track_crud_records(action="/me/search", record_or_dataset=dataset, count=search_responses.total)
```

davidberenstein1957 replied:
Sure, I will review this bahavior.

frascuchon commented on argilla-server/src/argilla_server/api/handlers/v1/datasets/records.py:
```diff
@@ -414,6 +423,8 @@ async def search_dataset_records(
             query_score=record_id_score_map[record.id]["query_score"],
         )
 
+    await telemetry_client.track_crud_records(action="read", record_or_dataset=dataset, count=search_responses.total)
```
```suggestion
    await telemetry_client.track_crud_records(action="search", record_or_dataset=dataset, count=search_responses.total)
```

davidberenstein1957 replied:
I will review this behavior

frascuchon commented on argilla-server/src/argilla_server/api/handlers/v1/records.py:
```diff
@@ -105,13 +115,18 @@ async def create_record_response(
-    return await datasets.create_response(db, search_engine, record, current_user, response_create)
+    response = await datasets.create_response(db, search_engine, record, current_user, response_create)
+
+    await telemetry_client.track_crud_records_subtopic(action="create", sub_topic="responses", record_id=record_id)
```
Why not just `track_crud_record_responses` and pass the record as argument?

davidberenstein1957 replied:
because I would need to create a lot of different methods to avoid code duplication but I will do so. Also, the record is not available everywhere so the record id is passed.

frascuchon commented on argilla-server/src/argilla_server/api/handlers/v1/records.py:
```diff
@@ -127,6 +142,10 @@ async def get_record_suggestions(
 
     await authorize(current_user, RecordPolicy.get_suggestions(record))
 
+    await telemetry_client.track_crud_records_subtopic(
```
better track_crud_record_suggestions

davidberenstein1957 replied:
because I would need to create a lot of different methods but I will do so

frascuchon commented on argilla-server/src/argilla_server/api/handlers/v1/workspaces.py:
```diff
@@ -88,6 +105,10 @@ async def list_workspaces_me(
 
+    for workspace in workspaces:
+        await telemetry_client.track_crud_workspace(action="read", workspace=workspace)
+    await telemetry_client.track_crud_workspace(action="read", workspace=None, count=len(workspaces))
```
I don't understand these lines. For other list endpoints we just track the resource count, but here we track also the workspaces individually. What's the motivation?

davidberenstein1957 replied:
@frascuchon It is to keep track of list like operations but not differentiate too much in the naming CRUD. If you think it is interesting for development, I will create a separate "list" action. Otherwise, I will leave it. I think the individual calls and list-like I forgot unintentionally.

davidberenstein1957 replied:
I reviewed the code and see I did keep track of the list-like thing in other places too.

frascuchon commented on argilla-server/src/argilla_server/errors/error_handler.py:
```diff
@@ -53,15 +53,14 @@ def __init__(self, error: ServerError):
-            data["type"] = error.type
 
-        telemetry.get_telemetry_client().track_data(action="ServerErrorFound", data=data)
+        await get_telemetry_client().track_data(topic="error/server", user_agent=user_agent)
```
Why not align with the other methods and create a `track_error(error, ...)` method?

davidberenstein1957 replied:
Because there is a single track error/server, but I will create a separate method for that.

frascuchon commented on argilla/docs/reference/telemetry.md:
should we also change/update the server configuration env vars section?

davidberenstein1957 replied:
Yes, we do. We initially did not have that section.

davidberenstein1957 committed:
docs: update telemetry sections

jfcalvo commented on .github/workflows/argilla-frontend.deploy-environment.yml:
```diff
@@ -63,7 +63,7 @@ jobs:
             ANNOTATOR_PASSWORD=${{ steps.credentials.outputs.annotator }}
             ANNOTATOR_API_KEY=${{ steps.credentials.outputs.annotator }}
-            ARGILLA_ENABLE_TELEMETRY=0
+            HF_HUB_DISABLE_TELEMETRY=1
```
It's easier to use affirmative settings:

```suggestion
            HF_HUB_ENABLE_TELEMETRY=0
```

In fact we already have an `ARGILLA_ENABLE_TELEMETRY` environment variable.

davidberenstein1957 replied:
@jfcalvo, I know but I think it is cleaner to default to the Hugging Face one because everything is stored there and to default to other HF-related configs people might have w.r.t. telemetry.

jfcalvo commented on argilla-server/pyproject.toml:
```diff
@@ -57,6 +57,8 @@ dependencies = [
     "packaging>=23.2",
     "psycopg2-binary>=2.9.9",
+    # For Telemetry
+    "huggingface_hub>=0.13,<1",
```
Why this specific range of versions?

davidberenstein1957 replied:
Because telemetry was introduced in 0.13, and I want to avoid going to another major release.

davidberenstein1957 committed:
update: usage from record_subtopic to record_suggestions and record_responses

jfcalvo commented on argilla-server/src/argilla_server/api/handlers/v1/datasets/datasets.py:
```diff
@@ -90,29 +90,57 @@ async def list_current_user_datasets(
+    for field in dataset.fields:
+        await telemetry_client.track_crud_dataset_setting(
+            action="read", dataset=dataset, setting_name="fields", setting=field
+        )
```
Do we know how these telemetry requests are done? Are they synchronous? UDP?

We should check that we are not spending a lot of time executing these requests so we are not adding additional time to the API endpoint requests.

davidberenstein1957 replied:
@jfcalvo underlying code can be found here. https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/utils/_telemetry.py

davidberenstein1957 committed:
refactor: introduced track_error specific method

jfcalvo commented on argilla-server/src/argilla_server/api/handlers/v1/records.py:
```diff
@@ -171,10 +191,16 @@ async def upsert_suggestion(
+    action = "create"
     if await Suggestion.get_by(db, record_id=record_id, question_id=suggestion_create.question_id):
         response.status_code = status.HTTP_200_OK
+        action = "update"
```
Maybe we can have an `upsert` action so you don't need to add logic trying to know if it's a `create` or an `update`?

davidberenstein1957 replied:
We can also rename it to "upsert" but I wanted to avoid capturing all edge cases "publish", "search", "read", "list" "upsert" "update" because I thought it might be a bit much for metrics/telemetry. @frascuchon @jfcalvo if you feel it would help development, I can make a finer distinguishment.

davidberenstein1957 committed:
refactor: name search operation like "search"
rafactor: add "me" to user operations
refactor: add "list" to like-like operations

frascuchon commented on argilla-server/src/argilla_server/telemetry.py:
```diff
@@ -16,115 +16,270 @@
+    def _process_dataset_settings(dataset: Dataset) -> dict:
+        user_data = {}
+        if dataset.is_relationship_loaded("guidelines"):
+            user_data["guidelines"] = True if getattr(dataset, "guidelines") else False
```
This value can be huge in some cases, we must be sure this won't impact the platform for those cases.

davidberenstein1957 replied:
I just share if they have been added or not, I don't store all of the guidelines.

frascuchon commented on argilla-server/src/argilla_server/telemetry.py:
```diff
@@ -16,115 +16,270 @@
-            context = self._system_info.copy()
+            user_agent.update(self._system_info)
+        if count is not None:
+            user_agent["count"] = count
```
what's the meaning of `count` and what is used for?

davidberenstein1957 replied:
Count is used for list operations

frascuchon replied:
Why not just send it as part of the `data` content for those  actions measuring list?

frascuchon committed:
[FEAT] argilla server: add basic endpoints telemetry support (#5437)

# Description
<!-- Please include a summary of the changes and the related issue.
Please also include relevant motivation and context. List any
dependencies that are required for this change. -->

This PR adds a middleware component to track the API endpoint's usage.

**Type of change**
<!-- Please delete options that are not relevant. Remember to title the
PR according to the type of change -->

- New feature (non-breaking change which adds functionality)
- Refactor (change restructuring the codebase without changing
functionality)
- Improvement (change adding some improvement to an existing
functionality)

**How Has This Been Tested**
<!-- Please add some reference about how your feature has been tested.
-->

**Checklist**
<!-- Please go over the list and make sure you've taken everything into
account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

---------

Co-authored-by: José Francisco Calvo <jose@argilla.io>

frascuchon committed:
Merge branch 'develop' into feat/5204-feature-add-huggingface_hubutilssend_telemetry-to-the-argilla-server

frascuchon committed:
chore: Remove all non-general endpoint telemetry related-code

frascuchon commented on argilla/mkdocs.yml:
```diff
@@ -184,7 +184,7 @@ nav:
         - Server configuration: reference/argilla-server/configuration.md
       - Telemetry:
-        - Server Telemetry: reference/argilla-server/telemetry.md
+        - Server Telemetry: reference/telemetry.md
```
```suggestion
        - Server Telemetry: reference/argilla-server/telemetry.md
```

frascuchon committed:
Update argilla/mkdocs.yml

frascuchon commented on argilla/docs/reference/argilla-server/telemetry.md:
```diff
@@ -45,10 +45,24 @@ The following usage and error information is reported:
 * The machine type, e.g. `AMD64`
 * The underlying platform spec with as much useful information as possible. (eg. `macOS-10.16-x86_64-i386-64bit`)
-* The type of deployment: `huggingface_space` or `server`
+* The type of deployment: `quickstart` or `server`, and if it is deployed on Hugging Face spaces.
```
```suggestion
* The type of deployment: `huggingface_space` or `server`
```

frascuchon committed:
chore: Revert doc change

frascuchon committed:
chore: revert doc changes

frascuchon committed:
chore: Remove unused attribute

frascuchon committed:
[FEAT] `argilla server`: add user and server id on telemetry metrics (#5445)

# Description
<!-- Please include a summary of the changes and the related issue.
Please also include relevant motivation and context. List any
dependencies that are required for this change. -->

This PR restores the server_id for telemetry purposes and also add the
user.id and user.role when tracking API requests.

**Type of change**
<!-- Please delete options that are not relevant. Remember to title the
PR according to the type of change -->

- Improvement (change adding some improvement to an existing
functionality)
- Documentation update

**How Has This Been Tested**
<!-- Please add some reference about how your feature has been tested.
-->

**Checklist**
<!-- Please go over the list and make sure you've taken everything into
account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

frascuchon committed:
[FEAT] `argilla server`: track servert startup (#5447)

# Description
<!-- Please include a summary of the changes and the related issue.
Please also include relevant motivation and context. List any
dependencies that are required for this change. -->

This PR adds the track startup method defined in
https://github.com/argilla-io/argilla/pull/5441 and include
perstitent_storaged_enbled info as part of the system info

**Type of change**
<!-- Please delete options that are not relevant. Remember to title the
PR according to the type of change -->

- Improvement (change adding some improvement to an existing
functionality)

**How Has This Been Tested**
<!-- Please add some reference about how your feature has been tested.
-->

**Checklist**
<!-- Please go over the list and make sure you've taken everything into
account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

frascuchon committed:
chore: Align the user.id registration

frascuchon committed:
chore: review docs

frascuchon committed:
chore: Update CHANGELOG
